scipy/io/arff/arffread.py

```diff
@@ -199,6 +197,12 @@ def get_date_format(atrv):
         if "ss" in pattern:
             pattern = pattern.replace("ss", "%S")
             datetime_unit = "s"
+        if "SSS" in pattern:
+            pattern = pattern.replace("SSS", "%f")
+            datetime_unit = "s"
```

Comment:
Oops, this is a bug and also demonstrates insufficient coverage. Amendment coming up soon…